Library/ViewModels/UpdateDraftViewModel.swift

```diff
@@ -187,17 +192,21 @@ UpdateDraftViewModelOutputs {
 
     let addAttachmentEvent = draft
       .takePairWhen(self.imagePickedProperty.signal.skipNil().map(first))
-      .switchMap { draft, url in
-        AppEnvironment.current.apiService.addImage(file: url, toDraft: draft)
+      .switchMap { (arg: (UpdateDraft, URL))
+        -> SignalProducer<Signal<UpdateDraft.Image, ErrorEnvelope>.Event, NoError> in
```

Comment:
Usually once you have the compiler happy with the return type then you shouldn't need to deconstruct the tuple.

Reply:
👍